src/Microsoft.SqlTools.ResourceProvider/ResourceCommandOptions.cs

```diff
@@ -0,0 +1,42 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Globalization;
+using Microsoft.SqlTools.Hosting.Utility;
+
+namespace Microsoft.SqlTools.ResourceProvider
+{
+    class ResourceCommandOptions : CommandOptions
+    {
+        internal const string ResourceServiceName = "SqlToolsResourceProviderService.exe";
+
+        public ResourceCommandOptions(string[] args) : base(args, ResourceServiceName)
+        {
+        }
+
+        public override void SetLocale(string locale)
+        {
+            try
+            {
+                LocaleSetter(locale);
+
+                // Setting our internal SR culture to our global culture
+                SR.Culture = CultureInfo.CurrentCulture;
+
+                // Setting Core SR culture to our global culture
+                Microsoft.SqlTools.ResourceProvider.Core.SR.Culture = CultureInfo.CurrentCulture;
```

Comment:
Is this really the only way to do this? It seems odd to be manually setting the culture like this ourself - typically C# programs get strings based on the current culture from satellite DLLs that are used based on the culture of the current thread. Are you following some pre-existing code here with doing it this way?

Reply:
This is meant for testing only, it works fine when using sqltoolsservice in azuredatastudio without it, but the tests do not work without it, as string resources are null instead of being properly translated when the SR.culture = culture line is removed.

This is following:
https://github.com/microsoft/sqltoolsservice/blob/alex/localizationbuild/src/Microsoft.Kusto.ServiceLayer/Utility/ServiceLayerCommandOptions.cs

and 

https://github.com/microsoft/sqltoolsservice/blob/alex/localizationbuild/src/Microsoft.SqlTools.ServiceLayer/Utility/ServiceLayerCommandOptions.cs